Src/AutoFixture.NUnit3/InlineAutoDataAttribute.cs

```diff
@@ -0,0 +1,112 @@
+        /// and parameter values for test method
+        /// </summary>
+        protected InlineAutoDataAttribute(IFixture fixture, params object[] arguments)
+        {
+            if (null == arguments)
+            {
+                throw new ArgumentNullException(nameof(arguments));
+            }
+
+            this._fixture = fixture;
+            this._existingParameterValues = arguments;
+        }
+
+        /// <summary>
+        ///     Construct one or more TestMethods from a given MethodInfo,
+        ///     using available parameter data.
+        /// </summary>
+        /// <param name="method">The MethodInfo for which tests are to be constructed.</param>
+        /// <param name="suite">The suite to which the tests will be added.</param>
+        /// <returns>One or more TestMethods</returns>
+        public IEnumerable<TestMethod> BuildFrom(IMethodInfo method, Test suite)
+        {
+            var test = new NUnitTestCaseBuilder().BuildTestMethod(method, suite, this.GetParametersForMethod(method));
+
+            yield return test;
+        }
+
+        private TestCaseParameters GetParametersForMethod(IMethodInfo method)
+        {
+            try
+            {
+                var parameters = method.GetParameters();
+
+                var parameterValues = this.GetParameterValues(parameters);
+
+                return new TestCaseParameters(parameterValues.ToArray());
+            }
+            catch (Exception ex)
+            {
+                return new TestCaseParameters(ex);
+            }
+        }
+
+        /// <summary>
+        /// Get values for a collection of <see cref="IParameterInfo"/>
+        /// </summary>
+        private IEnumerable<object> GetParameterValues(IEnumerable<IParameterInfo> parameters)
+        {
+            return this._existingParameterValues.Concat(this.GetMissingValues(parameters));
+        }
+
+        private IEnumerable<object> GetMissingValues(IEnumerable<IParameterInfo> parameters)
+        {
+            var parametersWithoutValues = parameters.Skip(this._existingParameterValues.Count());
+
+            return parametersWithoutValues.Select(this.GetValueForParameter);
+        }
+
+        /// <summary>
+        /// Get value for an <see cref="IParameterInfo"/>
+        /// </summary>
+        private object GetValueForParameter(IParameterInfo parameterInfo)
+        {
+            CustomizeFixtureByParameter(parameterInfo);
+
+            return new SpecimenContext(this._fixture)
+                .Resolve(parameterInfo.ParameterInfo);
+        }
+
+        private void CustomizeFixtureByParameter(IParameterInfo parameter)
+        {
+            var customizeAttributes = parameter.GetCustomAttributes<CustomizeAttribute>(false);
+            foreach (var ca in customizeAttributes)
+            {
+                var customization = ca.GetCustomization(parameter.ParameterInfo);
+                this._fixture.Customize(customization);
```

Comment:
Isn't the presence of an `IFixture` object a precondition for this class to work?